src/Compilers/Core/Portable/PEWriter/BlobUtilities.cs

```diff
@@ -0,0 +1,274 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using Roslyn.Utilities;
+
+namespace Microsoft.Cci
+{
+    internal unsafe static class BlobUtilities
+    {
+        public static void WriteBytes(this byte[] buffer, int start, byte value, int count)
+        {
+            fixed (byte* bufferPtr = buffer)
```

Comment:
If you can guarantee (assert) that buffer is non-null, then you can write this:
```byte* bufferPtr = &buffer[0]```
which avoids the implied null check. The JITted code should be smaller too.